shared.Makefile

```diff
@@ -1,5 +1,11 @@
 # This Makefile contains tasks and underlying pieces that are shared
 # across our other Makefiles.
+#
+# Build the Docker images used for CI tasks.
+#
+# The script sticks a record that the image has been built in .docker, so the
+# image isn't rebuilt unless you run 'make clean' first.  This makes CI tasks
+# slightly less chatty when run locally.
```

Comment:
👍 

useful comment

I can imagine pushing these some of them down into their individual projects as appropriate, but this is fine for now.